src/main/java/com/microsoft/azure/helium/app/HealthzController.java

```diff
@@ -79,21 +76,25 @@
         ieTfResult.put("version", buildConfig.getBuildVersion());
 
 
-        LinkedHashMap<String, HashMap<String, String>> healthChecks = new LinkedHashMap<>();
+        LinkedHashMap<String, LinkedHashMap<String, Object>> healthChecks = new LinkedHashMap<>();
         try {
             healthChecks.put("getGenres", runHealthChecksOnEndpoints("/api/genres", 400));
             healthChecks.put("getActorById", runHealthChecksOnEndpoints("/api/actors/nm0000173", 250));
             healthChecks.put("getMovieById", runHealthChecksOnEndpoints("/api/movies/tt0133093", 250));
             healthChecks.put("searchMovies", runHealthChecksOnEndpoints("/api/movies?q=ring", 400));
             healthChecks.put("searchActors", runHealthChecksOnEndpoints("/api/actors?q=nicole", 400));
-            healthChecks.put("getTopRatedMovie", runHealthChecksOnEndpoints("/api/movies?toprated=true", 400));
+            healthChecks.put("getTopRatedMovies", runHealthChecksOnEndpoints("/api/movies?toprated=true", 400));
 
-            for(Map.Entry<String, HashMap<String, String>> entry : healthChecks.entrySet()) {
-                HashMap check = entry.getValue();
-                if (check.containsKey(IeTfStatus.pass.name())) {
+            // if any health check has a warn or down status -  set overall status to the worst status
+            for(Map.Entry<String, LinkedHashMap<String, Object>> entry : healthChecks.entrySet()) {
+                LinkedHashMap check = entry.getValue();
+                if (check.containsValue(IeTfStatus.pass.name())) {
```

Comment:
I don't think this will work. If any of the status are warn, you should return warn. If the first status is warn and the last is pass, I believe this will return pass

Reply:
Yes, realized that, had to handle duration check outside of try/catch.
Done!